### DISCUSSION TITLE
Waycorner exec problem due to waybar

### DISCUSSION THREAD

**Discussion by mahaveergurjar** [prasanthrangan/hyprdots#507] [General]
I am using Waycorner for a hot corner but Waycorner is not working that side the way bar  
but waycorner is working on waybar side also when runnin waycorner in terminal but not working when implement in hyrland config exex-once

**Comment by mahaveergurjar** [comment]
https://github.com/prasanthrangan/hyprdots/assets/57872034/4ae15985-42bf-4e12-986a-20dabb3d9f3d

**Comment by kRHYME7** [comment]
hi, where do you install Waycorner from? AUR? I will try to replicate this erros

**Reply by mahaveergurjar** [reply]
AUR

**Reply by kRHYME7** [reply]
I see the problem now, as waybar  work as a layer also waycorner . I'll update you soon if I find something, this  utility looks very handy.

**Reply by mahaveergurjar** [reply]
Okay

**Reply by mahaveergurjar** [reply]
And please also find how to and script or shortcut ki in waybar

**Reply by kRHYME7** [reply]
Hi @mahaveergurjar  I got it working using exec-once but this is only a work around we needed  something to configure layers. Example is we would like to assume waybar to be at the bottom of waycorner. 

My workaround is launching waybar then delaying  waycorner . On my test device it took around 10 seconds to launch waybar so I decided to use 30 second delay to launch way corner.

so 
``` 
exec-once = sleep 30 && waycorner 

```
you can add this in your userprefs.


I'll try to do some research on how to configure this layers. Cheers

**Reply by kRHYME7** [reply]
>And please also find how to and script or shortcut ki in waybar

Can you please elaborate more? 

I'll try to play around with this utility because I find it very handy, thanks to you!

**Reply by mahaveergurjar** [reply]
I mean how to add bind key in waycorner or script like change waybar, wallpaper, volume, brightness etc those scripts how to add in waycorner

**Reply by kRHYME7** [reply]
I received a response from the dev just today, so i'll continue working on waycorner



I added "null" inside description so that it won't activate (This is not documented but it works)


#### Using the command
```
enter_command = [ "command", "flags parameters " ]
```

 + 1st field should be the vlild command in your $PATH  example ``` hyprctl ``` , ``` Hyprdots ``` etc

+ 2nd field should be the rest of the command. 



#####  Here's my config file. 

```
[top_left]
enter_command = [ "notify-send", "enter top left" ]
exit_command = [ "notify-send", "exit top left" ]
locations = ["top_left"] # default
size = 10 # default
margin = 20 # default
timeout_ms = 250 # default
color = "#FF0000" # Red
[top_left.output]
description = "null"  

[top_right]
enter_command = [ "notify-send", "enter top right" ]
exit_command = [ "notify-send", "exit top right" ]
locations = ["top_right"] # default
size = 10 # default
margin = 20 # default
timeout_ms = 250 # default
color = "#FFA500" # Orange
[top_right.output] 
description = "null"  

[bottom_left]
enter_command = [ "notify-send", "enter bottom left" ]
exit_command = [ "notify-send", "exit bottom left" ]
locations = ["bottom_left"] # default
size = 10 # default
margin = 20 # default
timeout_ms = 250 # default
color = "#FFFF00" # Yellow
[bottom_left.output]
description = "null"  

[bottom_right]
enter_command = [ "hyprctl", "dispatch workspace name: " ]
exit_command = [ "hyprctl", "dispatch workspace previous" ]
locations = ["bottom_right"] # default
size = 5 # default
margin = 20 # default
timeout_ms = 250 # default
color = "#008000" # Green
[bottom_right.output]
description = ""  

[top]
enter_command = [ "notify-send", "enter top" ]
exit_command = [ "notify-send", "exit top" ]
locations = ["top"] # default
size = 10 # default
margin = 20 # default
timeout_ms = 250 # default
color = "#0000FF" # Blue
[top.output]
description = "null"  

[bottom]
enter_command = [ "notify-send", "enter bottom" ]
exit_command = [ "notify-send", "exit bottom" ]
locations = ["bottom"] # default
size = 10 # default
margin = 20 # default
timeout_ms = 250 # default
color = "#4B0082" # Indigo
[bottom.output]
description = "null"  

[left]
enter_command = [ "hyprctl", "dispatch workspace r-1" ]
# exit_command = [ "notify-send", "exit left" ]
locations = ["left"] # default
size = 1 # default
margin = 100 # default
timeout_ms = 250 # default
color = "#EE82EE" # Violet
[left.output]
description = ""  

[right]
enter_command = [ "hyprctl", "dispatch workspace r+1" ]
# exit_command = [ "notify-send", "exit right" ]
locations = ["right"] # default
size = 1 # default
margin = 100 # default
timeout_ms = 250 # default
color = "#FF0000" # Red (repeated for demonstration)
[right.output]
description = ""  

```

**Comment by mahaveergurjar** [comment]
Are you using laptop ???

**Reply by kRHYME7** [reply]
Yes

**Reply by mahaveergurjar** [reply]
I need also touchpad gesture  for easy to use laptop like hot corner touchpad gesture are easy  my work like like to finger zoom in zoom out two finger is it work but only at browser not in all app like vscode kitty and etc and like four finger swipe down to take screen shot and like four finger left or right to brightness or volume up down and etc

**Comment by mahaveergurjar** [comment]
these all are possible through libinput-gestures but how to implement in hyprland i dont know 

viste these link i hope these are helpful for you
https://wiki.hyprland.org/Configuring/Keywords/#gestures
https://github.com/bulletmark/libinput-gestures

**Comment by kRHYME7** [comment]
@mahaveergurjar  Good news I hope this can be fixed soon. 

https://github.com/AndreasBackx/waycorner/issues/19

Stay tuned

**Reply by mahaveergurjar** [reply]
Wow 😀

**Reply by kRHYME7** [reply]
https://github.com/AndreasBackx/waycorner/pull/22

If you're comfortable, maybe you can test it

**Reply by kRHYME7** [reply]
@mahaveergurjar  Fixed it https://github.com/AndreasBackx/waycorner/commit/b68a5414a195bd791885405f26f5b6d5ee3b85cf

update your waycorner

**Comment by mahaveergurjar** [comment]
Thanks @kRHYME7
